source/baseObject.py

```diff
@@ -183,7 +183,7 @@ def invalidateCaches(cls):
 class ScriptableType(AutoPropertyType):
 	"""A metaclass used for collecting and caching gestures on a ScriptableObject"""
 
-	def __new__(cls, name: str, bases: tuple[type, ...], dict: [str, Any], /, **kwargs: Any):
+	def __new__(cls, name: str, bases: tuple[type, ...], dict: dict[str, Any], /, **kwargs: Any):
```

Comment:
Could we rename dict to dct while at it? dict is shadowing the type here:
```suggestion
	def __new__(cls, name: str, bases: tuple[type, ...], dct: dict[str, Any], /, **kwargs: Any):
```

Reply:
I don't think this is type shadowing, in fact this is a required parameter name for metaclasses `__new__`.

Reply:
Hmm, it's positional only, so the name doesn't matter, except to the method body. Do you have a reference?

Reply:
Ah I stand corrected. ABCMeta, which this is meant to shadow, uses `namespace` rather than `dict`